eventsource-client/examples/tail.rs

```diff
@@ -18,7 +19,7 @@ async fn main() -> Result<(), es::Error> {
     let url = &args[1];
     let auth_header = &args[2];
 
-    let client = es::Client::for_url(url)?
+    let client = es::for_url(url)?
```

Comment:
Do we think it would be better to make for_url a static method on the ClientBuilder struct? It seems weird to have this one standalone function as in the ingress point into all this functionality.